### DISCUSSION TITLE
How to install the binary in Laravel Vapor with Docker

### DISCUSSION THREAD

**Discussion by Qalaha** [spatie/pdf-to-text#60] [General]
The following info could be added to Requirements section in README.md :-)
To install the binary on Alpine Linux used by Laravel Vapor with Docker add this to your Dockerfile (e.g. production.Dockerfile):
```
RUN apk update
RUN apk add --upgrade poppler-utils
```
(as mentioned here: https://alpine.pkgs.org/3.14/alpine-main-x86_64/poppler-utils-21.06.1-r0.apk.html)

For use in your local Laravel Docker container (Laravel Sail) you can use 
```
RUN apt-get install -y poppler-utils
```

**Comment by jtomek** [comment]
This is very helpful. Thank you.

**Comment by azam1** [comment]
To add the installation of `pdftotext` to your Laravel Sail setup, you should modify the `Dockerfile` referenced in your `docker-compose.yml` file. In your `docker-compose.yml`, there's a section for the `laravel.test` service that specifies a build context and a Dockerfile located at `./docker/8.3/Dockerfile`.

Here's what you need to do:

1. **Locate the Dockerfile**: Go to the `./docker/8.3/` directory in your Laravel project and find the `Dockerfile`.

2. **Edit the Dockerfile**: Open the `Dockerfile` and add the installation command for `pdftotext`. You would typically add these lines:
   ```Dockerfile
   RUN apt-get update && apt-get install -y poppler-utils
   ```
   This command updates the package list and installs `poppler-utils`, which includes `pdftotext`.

3. **Rebuild Your Docker Containers**: After saving your changes to the Dockerfile, go back to the root of your Laravel project and run `./vendor/bin/sail build` to rebuild the Docker containers with the new changes.

4. **Restart Laravel Sail**: Finally, restart Laravel Sail to apply the changes using `./vendor/bin/sail up -d`.

After these steps, `pdftotext` should be installed in your Laravel Sail environment.

**Comment by msucevan** [comment]
@Qalaha hi, thank you for the following instructions. I did the same in my project. 

But I have a question, did you get the file from the storage and save it in the tmp files, then you can set the PDF?

Since we are in the serverless environment, I don't understand exactly what is the best to get the pdf from a S3 storage
